### ISSUE TITLE
Share your AI ideas!

### ISSUE THREAD

**Issue by hakangoksu** [end-4/dots-hyprland#754]
Connecting/using openwebui in the left bar should be the perfect thing. I want to add my own models that are working in my pc and openwebui is perfect for it. It can download-start ollama models. Integrating with your sidebar would be perfect.

My idea is:

Making 4 section just 2 main section. Online and local

In online using api keys etc. 

In local section integrating with openwebui. Changing between our models -presets 

Designing, I believe you guys make the best. So please let me know what are you thinking about these ideas? If I can help, I would gladly do it as much as I can also.

**Comment by hakangoksu** [comment]
Yes, the thing with that API's I can't access literally nothing. Gemini doesn't provide api's in my country and chatgpt doesnt work for some reason. In anycase chatgpt is not free. I like to use my own ai's. And they don't require toooo high specs btw. You can get gpt3.0 level with ~8 gb ram like the one I'm using is llama3.1. There are also model that work ~2gb even. I know it is not very good but there are models for any pc specs in ollama where the openwebui connects to.

I would really like to use my ai models integrated to this end-4's hyprland.

**Comment by braveltd** [comment]
> Yes, the thing with that API's I can't access literally nothing. Gemini doesn't provide api's in my country and chatgpt doesnt work for some reason. In anycase chatgpt is not free. I like to use my own ai's. And they don't require toooo high specs btw. You can get gpt3.0 level with ~8 gb ram like the one I'm using is llama3.1. There are also model that work ~2gb even. I know it is not very good but there are models for any pc specs in ollama where the openwebui connects to.
> 
> I would really like to use my ai models integrated to this end-4's hyprland.

same thing for me. I live in russia, and there's gemini, chatgpt, other's aren't working without vpn.
i use openrouter and oxygen (which is limited). so would like to see some vpn for end-4 dots

**Comment by Kalidude69** [comment]
> Yes, the thing with that API's I can't access literally nothing. Gemini doesn't provide api's in my country and chatgpt doesnt work for some reason. In anycase chatgpt is not free. I like to use my own ai's. And they don't require toooo high specs btw. You can get gpt3.0 level with ~8 gb ram like the one I'm using is llama3.1. There are also model that work ~2gb even. I know it is not very good but there are models for any pc specs in ollama where the openwebui connects to.
> 
> 
> 
> I would really like to use my ai models integrated to this end-4's hyprland.

Alright you also have a point tho!!

**Comment by hakangoksu** [comment]
Okay, I managed to do some stuff by myself. I want to update and contribute to the project also. 

For now here is what you can do: ~/.config/ags/services/gps.js

In the provider section, you can change order and add your own

Here is an example for llama3.1:8b (since I dont want to use llama3:instruct)

```
'ollama3.1': { //Change title, I think its important since its defined here
        'name': 'Ollama (Llama 3.1)', //You can change name. This will display on the bar
        'logo_name': 'ollama-symbolic',
        'description': 'Official Ollama API.\nPricing: Free.', //You can add description
        'base_url': 'http://localhost:11434/v1/chat/completions', 
        'key_get_url': 'it\'s just ollama',
        'key_file': 'ollama_key.txt', 
        'model': 'llama3:instruct', //
    },
```

I'm still trying to understand how this dost-hyprland made the ai system. If I can do something useful by myself I want to contribute to the project if people likes..

**Comment by andresparrab** [comment]
I would love to see chat gpt integrated. I have in in the terminal, in Jetbrains Rider, browser and more.. but on the side woul be anice companion :)

**Comment by Kalidude69** [comment]
> I would love to see chat gpt integrated. I have in in the terminal, in Jetbrains Rider, browser and more.. but on the side woul be anice companion :)

sir there is openai on the side panel for that you would need to pay for the api usage..

**Comment by Kalidude69** [comment]
> oh of course..sorry it was dumb of me i saw it now

it's fine no need to be sorry :D

**Comment by pandalec** [comment]
It would be nice, if I could easily disable all unused AI providers. And it would be also nice if AGS could read env variables as most of AI tools which I use already read the key from an env var

**Comment by end-4** [comment]
> but how about integrate bling copilot..it is really god and github copilot that would be great :)

Can't, there's no api. I'm not doing scraping stuff, it's just painful
Of course Microsoft is keeping it to themself

**Comment by end-4** [comment]
> It would be nice, if I could easily disable all unused AI providers. And it would be also nice if AGS could read env variables as most of AI tools which I use already read the key from an env var

doable
mind sharing what the env var names are?

**Comment by pandalec** [comment]
> > It would be nice, if I could easily disable all unused AI providers. And it would be also nice if AGS could read env variables as most of AI tools which I use already read the key from an env var
> 
> doable mind sharing what the env var names are?

Sure, thanks for looking into this

```
OPENAI_API_KEY=key
OPENAI_MODEL=gpt-4o-mini # gpt-4o
DEFAULT_MODEL=gpt-4o-mini # gpt-4o
OPENAI_MAX_TOKENS=16384
HANDLER=openai # openai/copilot/codeium
```

I have these configured for [shell-gpt](https://github.com/TheR1D/shell_gpt) and [helix-gpt](https://github.com/leona/helix-gpt)

**Comment by iceking-1912** [comment]
hey how can we integrate anime apps or cli like ani-cli in sider

**Comment by Rkastik08** [comment]
ok so it would be great if image generative apis can be changed to preffered one. and custom prompt giving for image generation will also be very useful.
